### PR TITLE
HHH-10837 - Add new setting to disable/enable scanning for the static metamodel

### PR DESCRIPTION
**Pull request by dreab8**
https://hibernate.atlassian.net/browse/HHH-10837

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/metamodel/internal/JpaStaticMetaModelPopulationSetting.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.metamodel.internal;
+
+import java.util.Map;
+
+import org.hibernate.cfg.AvailableSettings;
+import org.hibernate.internal.util.config.ConfigurationHelper;
+
+/**
```
Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/metamodel/internal/JpaStaticMetaModelPopulationSetting.java]: Can we add some javadocs here describing the purpose of this enumerated setting?  This seems the logical place to centralize its discussion.

Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/metamodel/internal/JpaStaticMetaModelPopulationSetting.java]: aka:
````
/**
 * Enumerated setting used to control whether Hibernate looks for and populates
 * JPA static metamodel models of application's domain model.
 */
````

File: hibernate-core/src/main/java/org/hibernate/metamodel/internal/JpaStaticMetaModelPopulationSetting.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ * @author Andrea Boriero
+ */
+public enum JpaStaticMetaModelPopulationSetting {
+	ENABLED,
```
Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/metamodel/internal/JpaStaticMetaModelPopulationSetting.java]:
````
/**
 * ENABLED indicates that Hibernate will look for the JPA static metamodel description 
 * of the application domain model and populate it.
 */
````

File: hibernate-core/src/main/java/org/hibernate/metamodel/internal/JpaStaticMetaModelPopulationSetting.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+	DISABLED,
```
Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/metamodel/internal/JpaStaticMetaModelPopulationSetting.java]: etc :)

**Comment by sebersole** [comment]
Aside from the minor line comments about javadoc, I think this looks good.  I don wonder a little about MetadataContext happening completely conditionally since we have merged HEM and core.  Did all of the tests pass with this change @dreab8, not just core?

**Comment by dreab8** [comment]
Hi @sebersole,
 
I run all the test suite, no failure.
With STATIC_METAMODEL_POPULATION = "disabled" only the JPA tests using static metamodel references fail while JPA_METAMODEL_POPULATION = "disable" causes the failure of all the tests using Criteria as it seems happening before the change.

I'm going to add the javadoc as you suggested.

Thanks for the review

**Comment by sebersole** [comment]
Then this looks good to me.

**Comment by dreab8** [comment]
applied upstream

**Pull request closed without merging**